Gems/Atom/Feature/Common/Code/Source/AuxGeom/AuxGeomDrawQueue.cpp

```diff
@@ -659,16 +659,17 @@ namespace AZ
 
             AuxGeomIndex vertexOffset = aznumeric_cast<AuxGeomIndex>(primBuffer.m_vertexBuffer.size());
             AuxGeomIndex indexOffset = aznumeric_cast<AuxGeomIndex>(primBuffer.m_indexBuffer.size());
+            const size_t vertexCountTotal = aznumeric_cast<size_t>(vertexOffset) + vertexCount;
 
-            if (aznumeric_cast<size_t>(vertexOffset) + vertexCount > MaxDynamicVertexCount)
+            if (vertexCountTotal > MaxDynamicVertexCount)
             {
                 AZ_WarningOnce("AuxGeom", false, "Draw function ignored, would exceed maximum allowed index of %d", MaxDynamicVertexCount);
                 return;
             }
 
             AZ::Vector3 center(0.0f, 0.0f, 0.0f);
-            primBuffer.m_vertexBuffer.reserve(vertexCount);
-            primBuffer.m_indexBuffer.reserve(vertexCount);
+            primBuffer.m_vertexBuffer.reserve(vertexCountTotal);
+            primBuffer.m_indexBuffer.reserve(vertexCountTotal);
```

Comment:
I haven't gone over the logic here but is that not a mistake?
primBuffer.m_indexBuffer.reserve(vertexCountTotal)

Reply:
This function (DrawPrimitiveCommon) doesn't take an indexCount as input, only a vertexCount, so I think for this one it's just assuming it's not actually indexed. The other (DrawPrimitiveWithSharedVerticesCommon) does, so the logic is correct to use the total vertex count here, and use the total index count in the other.

Reply:
you are correct, just looked at the inner loop, dropping